### PR TITLE
Dev spiffe TLS listen

### PR DESCRIPTION
**Pull request by otaviolcs3**

### CODE REVIEW

File: spiffetls/tests/check_listen.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+void *call_client(void *unused)
+{
+    sleep(1);
+    system("./tls_client 20001 &");
+    pthread_exit(NULL);
+    return NULL;
+}
+
+START_TEST(test_spiffetls_ListenWithMode)
```
Comment (arianalima) [inline spiffetls/tests/check_listen.c:13]: What do you think about commenting the test details like precondition and post condition?

File: spiffetls/tlsconfig/src/config.c [hunk 2]
```diff
@@ -132,6 +132,40 @@ bool tlsconfig_HookMTLSClientConfig(SSL_CTX *ctx, x509svid_Source *svid,
...
+    safe_arg->bundle = bundle;
+    safe_arg->authorizer = authorizer;
+    safe_arg->opts = opts;
+
+    SSL_CTX_set_cert_verify_callback(ctx, hookTLSClientConfig_cb, safe_arg);
+}
+
 void tlsconfig_resetAuthFields(SSL_CTX *ctx)
 {
     // config.Certificates = nil - ok
```
Comment (arianalima) [inline spiffetls/tlsconfig/src/config.c:171]: Should this commented block be here? (171 ~177)

> // config.Certificates = nil - ok
    // config.ClientAuth = tls.NoClientCert
    // config.GetCertificate = nil - ok
    // config.GetClientCertificate = nil - ok
    // config.InsecureSkipVerify = false - ok
    // config.NameToCertificate = nil //nolint:staticcheck // setting to nil is
    // OK config.RootCAs = nil

Comment (otaviolcs3) [reply spiffetls/tlsconfig/src/config.c:171]: Yes

**Review by rodrigolc — APPROVED** [review]

File: spiffetls/tests/check_dial.c [hunk 3]
```diff
@@ -23,7 +23,7 @@ START_TEST(test_spiffetls_DialWithMode)
     SSL *conn = spiffetls_DialWithMode((in_port_t) 40001,
                                        /*127.0.0.1*/ (in_addr_t) 0x7F000001,
                                        mode, &config, &err);
-
+
```
Comment (willallves) [inline spiffetls/tests/check_dial.c:26]: Remove this change.

File: spiffetls/tests/check_listen.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
+#include "spiffetls/src/listen.h"
+#include <check.h>
+#include <unistd.h>
+
+void *call_client(void *unused)
+{
+    sleep(1);
+    system("./tls_client 20001 &");
+    pthread_exit(NULL);
+    return NULL;
```
Comment (willallves) [inline spiffetls/tests/check_listen.c:10]: void function, no return.

File: spiffetls/tests/check_dial.c (hunk 3 above)
Comment (otaviolcs3) [inline spiffetls/tests/check_dial.c:26]:
```suggestion

```

File: spiffetls/tests/check_listen.c (hunk 4 above)
Comment (otaviolcs3) [reply spiffetls/tests/check_listen.c:10]: The return is void*, not void.

**Review by rodrigolc — APPROVED** [review]

**Review by willallves — APPROVED** [review]

**Review by arianalima — APPROVED** [review]

**Pull request merged**